src/NuGet.Clients/NuGet.VisualStudio.Contracts/NuGetInstalledPackage.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.VisualStudio.Contracts
+{
+    /// <summary>Information about an installed package</summary>
+    public sealed class NuGetInstalledPackage
+    {
+        /// <summary>The package id.</summary>
+        public string Id { get; }
+
+        /// <summary>The project's requested package range for the package.</summary>
+        /// <remarks>
+        /// If the project uses packages.config, this will be same as the installed package version.
+        /// If the project uses PackageReference, this is the version string in the project file, which may not match the resolved package version, and may not be single version string.
+        /// </remarks>
+        public string RequestedRange { get; }
+
+        /// <summary>The lowest version allowed by the <see cref="RequestedRange"/></summary>
+        public string RequestedVersion { get; }
+
+        // This class will hopefully use C# record types when that language feature becomes available, so make the constructor not-public, to prevent breaking change when records come out.
```

Comment:
I think we probably need the installed path here.

Reply:
I need to find the link you shared with me earlier to see how difficult it is to add. But install path is not used in any of the top UI delay root causes, so I think can be added later.

Reply:
It's super easy to add: https://github.com/NuGet/NuGet.Client/blob/72fb77af5c0c5dba95296584356173d594373804/src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstallerServices.cs#L68-L105

We can add it in 16.8. Think we want TestExplorer to use this ASAP too, even if it's not the TOP UI delay.

Reply:
thanks for helping me not need to find the other place you shared the same info :)  Expect it in an update.